### ISSUE TITLE
[BUG] During Speed Test boat stalls at far waypoint

### ISSUE THREAD

**Issue by cooperbell** [UIOWA5830SP19/SPP100#40]
**Describe the bug**
During the speed test the boat stalls for about ~10 seconds when it reaches the waypoint at the far buoy. 

**To Reproduce**
1. Run Dashboard
2. Run boat code with the speed test loaded
3. Wait for boat to get to the far buoy during the speed test

**Expected behavior**
The boat shouldn't take that long to go around the far buoy. It makes sense for it to stall for a short time (possibly to set up the next waypoint and get rid of the one it just got to) but not as long as it's taking

**Screenshots**

**Boat code or simulation code?**
Simulation

**Windows or Mac?**

**Additional context**

**Comment by cooperbell** [comment]
When the boat gets near the waypoint, logging the right & left thrust values shows behavior like this:

```
[2019/02/25 04:02:49.816 PM] controls_d: Left Thrust: 0.8611704053379542, Right Thrust: -0.11816759008189293
[2019/02/25 04:02:49.838 PM] controls_d: Left Thrust: 0.8613279085149574, Right Thrust: -0.09916334172383195
[2019/02/25 04:02:49.858 PM] controls_d: Left Thrust: 1.0, Right Thrust: -1.0
[2019/02/25 04:02:49.878 PM] controls_d: Left Thrust: 1.0, Right Thrust: -1.0
...
[2019/02/25 04:02:54.177 PM] controls_d: Left Thrust: 1.0, Right Thrust: -1.0
[2019/02/25 04:02:54.196 PM] controls_d: Left Thrust: 0.9949505772289396, Right Thrust: -0.9867032090156637
...
[2019/02/25 04:02:56.856 PM] controls_d: Left Thrust: 0.9009643659627204, Right Thrust: 0.43663537967504396
[2019/02/25 04:02:56.874 PM] controls_d: Left Thrust: 1.0, Right Thrust: -1.0
[2019/02/25 04:02:56.895 PM] mapping_v: Reached waypoint: (-91.5293, 41.6224, 0.0000)
[2019/02/25 04:02:56.896 PM] controls_d: Left Thrust: 1.0, Right Thrust: -1.0
...
```

**Comment by jwryan9** [comment]
The boat is not actually stalling as it approaches the far waypoint. It is making fine course adjustments to avoid entering the set buoy collision radius. By moving waypoint further from buoy the boat no longer "stalls." Currently testing time it takes for task completion at a variety of waypoint locations; will update.

**Comment by jwryan9** [comment]
Waypoint:
`{
      "x": -91.529310,
      "y": 41.622447,
      "size": 1
}`
Result:
```
[2019/02/26 03:40:58.811 PM] mapping_v: Reached waypoint: (-91.5287, 41.6224, 0.0000)
[2019/02/26 03:41:37.853 PM] mapping_v: Reached waypoint: (-91.5293, 41.6224, 0.0000)
[2019/02/26 03:42:16.952 PM] mapping_v: Reached waypoint: (-91.5287, 41.6224, 0.0000)
```
Difference:
`0:01:18.141`

**Comment by cooperbell** [comment]
@jwryan9 and I have been investigating this and have found that the code works as expected, just it's not ideal. The code can be optimized so the boat can deal with approaching waypoints more smoothly. Recommend opening a feature request for implementing this optimization.

What the code does now:
* The boat's GPS coordinates needs to be at the _exact_ GPS coordinates of the waypoint, so when the boat gets super close to the waypoint it spends a long time trying to fine tune the thrust to get there.

What the code should be optimized to:
* We should treat the waypoint as a type **collision object** in the same fashion that buoys are treated as collision objects, except instead of the boat avoiding this area, it'll be treated as an acceptance zone, so the boat only needs to get close to the waypoint's GPS point for it to count as getting there.

**Comment by jwryan9** [comment]
Waypoint:
`{
      "x": -91.529315,
      "y": 41.622447,
      "size": 1
}`
Result:
```
[2019/02/26 03:51:29.803 PM] mapping_v: Reached waypoint: (-91.5287, 41.6224, 0.0000)
[2019/02/26 03:52:10.914 PM] mapping_v: Reached waypoint: (-91.5293, 41.6224, 0.0000)
[2019/02/26 03:52:48.992 PM] mapping_v: Reached waypoint: (-91.5287, 41.6224, 0.0000)
```
Difference:
`0:01:19.189`

**Comment by jwryan9** [comment]
Waypoint:
`{
      "x": -91.529313,
      "y": 41.622447,
      "size": 1
}`
Result:
```
[2019/02/26 03:54:42.960 PM] mapping_v: Reached waypoint: (-91.5287, 41.6224, 0.0000)
[2019/02/26 03:55:25.067 PM] mapping_v: Reached waypoint: (-91.5293, 41.6224, 0.0000)
[2019/02/26 03:56:03.170 PM] mapping_v: Reached waypoint: (-91.5287, 41.6224, 0.0000)
```
Difference:
`0:01:20.21`

**Comment by jwryan9** [comment]
Waypoint:
`{
      "x": -91.529313,
      "y": 41.622447,
      "size": 1
}`

Result:
```
[2019/02/26 04:03:15.994 PM] mapping_v: Reached waypoint: (-91.5287, 41.6224, 0.0000)
[2019/02/26 04:03:58.099 PM] mapping_v: Reached waypoint: (-91.5293, 41.6224, 0.0000)
[2019/02/26 04:04:36.156 PM] mapping_v: Reached waypoint: (-91.5287, 41.6224, 0.0000)
```
Difference:
`0:01:20.162`

**Comment by jwryan9** [comment]
Based on these results will leave the waypoint at `{ "x": -91.529311, "y": 41.622447, "size": 1 }` for time being.

**Comment by cooperbell** [comment]
This will be closed and feature request #73 has been opened.
